### PR TITLE
Dependency update March 2026

### PR DESCRIPTION
**Pull request by KCeh**
## Summary

I update dependecies and related configs in project

## Changes

- compile and target SDK bump
- AGP bump to 9.1.0 😎 
- Fixed dokka config
- Removed ktlint to avoid conflicts with detekt (CI workflow also updated)
- Bumped dependecies in general
- Edge to edge support fixed in sample app

### Type

- [ ] **Feature**: This pull request introduces a new feature.
- [ ] **Bug fix**: This pull request fixes a bug.
- [x] **Refactor**: This pull request refactors existing code.
- [ ] **Documentation**: This pull request updates documentation.
- [ ] **Other**: This pull request makes other changes.

#### Additional information

- [ ] This pull request introduces a **breaking change**.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have tested my changes, including edge cases.
- [ ] I have added necessary tests for the changes introduced (if applicable).
- [ ] I have updated the documentation to reflect my changes (if applicable).

## Additional notes

I did check sample app, static checks and test. But it would also be good for one more pair of eyes to validate that.
Also, I am not planning to publish new version. But, before doing that we should also make sure publishing is ok on maven local.

### CODE REVIEW

**Comment by KCeh** [comment]
I am not sure what is going on with Github actions, look like network issue. I will retry later

File: sample/build.gradle [hunk 1]
```diff
@@ -12,7 +12,6 @@ buildscript {
 
 plugins {
     alias libs.plugins.android.application
```
[the remark this answers is not in the document]
Comment (KCeh) [reply sample/build.gradle:14]: It is compiling, this is AGP 9.0+ change

**Review by kristinakoneva — APPROVED** [review]

**Comment by KCeh** [comment]
I rechecked publishing to mavenLocal and it looked good to me
Since we don't have active use case for this lib there is no need to publish new version right away. We we will publish we can recheck if all is ok once again.
